src/decode/decode_gps_l2c.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Adel Mamin <adel.mamin@exafore.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include "decode_gps_l2c.h"
+#include "decode.h"
+
+#include <libswiftnav/logging.h>
+#include <libswiftnav/nav_msg.h> /* For BIT_POLARITY_... constants */
+#include <libswiftnav/cnav_msg.h>
+#include <assert.h>
+
+#include "ephemeris.h"
+#include "track.h"
+#include "sbp.h"
+#include "sbp_utils.h"
+#include "signal.h"
+
+/** GPS L2C CNAV message length in symbols */
+#define GPS_CNAV_MSG_LENGTH         (300)
+
+/** GPS L2C symbol length [ms] */
+#define GPS_L2C_SYMBOL_LENGTH          20
+
+/** Maximum probability of symbol 0 as
+    expected by cnav_msg_decoder_add_symbol(). */
+#define MAX_PROBABILITY_OF_SYMBOL_0        0
+
+/** Maximum probability of symbol 1 as
+    expected by cnav_msg_decoder_add_symbol(). */
+#define MAX_PROBABILITY_OF_SYMBOL_1        0xFF
+
+typedef struct {
+  cnav_msg_t cnav_msg;
+  cnav_msg_decoder_t decoder;
```

Comment:
Can we call this `cnav_msg_decoder` so it's a little less confusing? There are already a lot of decoders floating around.